host/libhackrf/src/hackrf.h

```diff
@@ -136,305 +778,939 @@ enum operacake_ports {
+ * @return @ref HACKRF_SUCCESS on success or @ref hackrf_error variant
+ * @ingroup configuration
+ */
 extern ADDAPI int ADDCALL hackrf_set_antenna_enable(
 	hackrf_device* device,
 	const uint8_t value);
 
+/**
+ * Convert @ref hackrf_error into human-readable string
+ * @param errcode enum to convert
+ * @return human-readable name of error
+ * @ingroup library
+ */
 extern ADDAPI const char* ADDCALL hackrf_error_name(enum hackrf_error errcode);
 
+/**
+ * Convert @ref hackrf_board_id into human-readable string
+ * 
+ * @param board_id enum to convert
+ * @return human-readable name of board id
+ * @ingroup device
+ */
 extern ADDAPI const char* ADDCALL hackrf_board_id_name(enum hackrf_board_id board_id);
 
+/**
+ * Lookup platform ID (HACKRF_PLATFORM_xxx) from board id (@ref hackrf_board_id)
+ * 
+ * @param board_id @ref hackrf_board_id enum variant to convert
+ * @return @ref HACKRF_PLATFORM_JAWBREAKER, @ref HACKRF_PLATFORM_HACKRF1_OG, @ref HACKRF_PLATFORM_RAD1O, @ref HACKRF_PLATFORM_HACKRF1_R9 or 0
+ * @ingroup device
+*/
 extern ADDAPI uint32_t ADDCALL hackrf_board_id_platform(enum hackrf_board_id board_id);
 
+// part of docstring is from hackrf.h
+/**
+ * Convert @ref hackrf_usb_board_id into human-readable string.
+ * @param usb_board_id enum to convert
+ * @return human-readable name of board id
+ * @ingroup device
+ */
 extern ADDAPI const char* ADDCALL hackrf_usb_board_id_name(
 	enum hackrf_usb_board_id usb_board_id);
 
+/**
+ * Convert @ref rf_path_filter into human-readable string
+ * @param path enum to convert
+ * @return human-readable name of filter path
+ * @ingroup configuration
+ */
 extern ADDAPI const char* ADDCALL hackrf_filter_path_name(const enum rf_path_filter path);
 
-/* Compute nearest freq for bw filter (manual filter) */
+/**
+ * Compute nearest valid baseband filter bandwidth lower than a specified value
+ * 
+ * The result can be used via @ref hackrf_set_baseband_filter_bandwidth
+ * 
+ * @param bandwidth_hz desired filter bandwidth in Hz
+ * @return the highest valid filter bandwidth lower than @p bandwidth_hz in Hz 
+ * @ingroup configuration
+ */
 extern ADDAPI uint32_t ADDCALL hackrf_compute_baseband_filter_bw_round_down_lt(
 	const uint32_t bandwidth_hz);
 
-/* Compute best default value depending on sample rate (auto filter) */
+/**
+ * Compute nearest valid baseband filter bandwidth to specified value
+ * 
+ * The result can be used via @ref hackrf_set_baseband_filter_bandwidth
+ * 
+ * @param bandwidth_hz desired filter bandwidth in Hz
+ * @return nearest valid filter bandwidth in Hz
+ * @ingroup configuration
+ */
 extern ADDAPI uint32_t ADDCALL hackrf_compute_baseband_filter_bw(
 	const uint32_t bandwidth_hz);
 
 /* All features below require USB API version 0x1002 or higher) */
 
-/* set hardware sync mode  */
+/**
+ * Set hardware sync mode (hardware triggering)
+ * 
+ * See the documentation on hardware triggering for details
+ * 
+ * Requires USB API version 0x1002 or higher!
```

Comment:
There's a typo in this and several other comments about required USB API versions.

The USB API version required here is represented as `0x0102`, not `0x1002`. In any case, since the documentation is for human consumption I think it would be better to say `1.02`. 

These comments are also inconsistent: they're present for several functions that require API 1.02, but there's no comments for other functions that require later API versions. For instance `hackrf_get_m0_state` requires API 1.06. I think we should either ensure required USB API versions are fully listed, or not mention them.

Reply:
I will do a review of USB API versions in a different commit later.